cmd/lima-guestagent/daemon_linux.go

```diff
@@ -72,10 +68,14 @@ func daemonAction(cmd *cobra.Command, _ []string) error {
 	r := mux.NewRouter()
 	server.AddRoutes(r, backend)
 	srv := &http.Server{Handler: r}
+	err = os.RemoveAll(socket)
+	if err != nil {
+		return err
+	}
 
 	var l net.Listener
-	if _, err := os.Stat(virtioPort); err == nil {
-		qemuL, err := serialport.Listen(virtioPort)
+	if virtioPort != "" {
```

Comment:
Instead of using virtioPort as a arg, can we simply do os.Stat and fallback to unix sock ?

Reply:
Having it as an explicit argument was a feature, but it should probably be named serialSock

Reply:
The communication medium is internal to us. I don't think there is a need to make it a argument. We can hardcode it like we do for unix sock just to simplify stuff

Reply:
If the filename is indeed fixed we can hardcode it, but then it might not even need the constant